File: src/main/java/com/mega/haksamate/entites/Subject.java
package com.mega.haksamate.entites;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Subject {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;
    private String dayOfWeek;  // 예: "MONDAY"
    private String startTime;  // 예: "09:00"
    private String endTime;    // 예: "10:30"
    private int priority;
}

File: frontend/src/types/subject.ts
// types/subject.ts
export interface Subject {
    id?: number;
    name: string;
    dayofweek: string;    // DB 컬럼과 동일!
    starttime: string;
    endtime: string;
    required: boolean;
    user_id?: string;
}

export interface TimetableSlot {
    dayofweek: string;
    starttime: string;
    endtime: string;
    subject: Subject;
}
